### ISSUE TITLE
Use other datasets to train PENet

### ISSUE THREAD

**Issue by yuyu19970716** [JUGGHM/PENet_ICRA2021#46]
Hello
I now want to use the PENet network architecture to complete depth completion, because I only need the dataset at hand, so I found the HandNet dataset, which contains depth data and image data collected by the realsense series of depth cameras. I especially want to know where do I need to modify the code? Thank you in advance!
Looking forward to your reply!

**Comment by JUGGHM** [comment]
Thanks for your interest yuyu! Individually I guess you could write your own dataloader for the HandNet dataset.

**Comment by yuyu19970716** [comment]
Thanks for your quick reply! If I write my own data loader for HandNet, if its depth map and rgb image are already aligned, do I need to use the function load_calib() in kitti_loader? I think I should not need the value of k. Very much looking forward to your reply!

**Comment by JUGGHM** [comment]
> Thanks for your quick reply! If I write my own data loader for HandNet, if its depth map and rgb image are already aligned, do I need to use the function load_calib() in kitti_loader? I think I should not need the value of k. Very much looking forward to your reply!

You might need it if you want to use the proposed geometric convolutional layer, otherwise not.

**Comment by yuyu19970716** [comment]
Thank you very much for your reply!
    Thank you very much for your reply!
    First of all, I may need to bother you with another question. Is K in the code the internal parameter matrix of the camera?
    In addition, I now want to train a 640*480 uncropped image dataset, I see that your code is 352*1216, so can I directly modify these two numbers?
Looking forward to your reply, thank you in advance! ! !

**Comment by yuyu19970716** [comment]
**Runtimeerror:The size of tensor a(288) must match the size of tensor b(1216) at non-singleton dimension3**


I have this problem now, do you know what is the reason? thank you very much!

**Comment by yuyu19970716** [comment]
Thank you very much for your reply!

I may need to bother you with another question. Is K in the code the internal parameter matrix of the camera?

Looking forward to your reply, thank you in advance! ! !

**Comment by JUGGHM** [comment]
> Thank you very much for your reply!
> 
> I may need to bother you with another question. Is K in the code the internal parameter matrix of the camera?
> 
> Looking forward to your reply, thank you in advance! ! !

Exactly.

**Comment by yuyu19970716** [comment]
alright, thank you very much! I trained the network today. I used the NYU-Depth v2 dataset and changed k to the camera's internal parameter matrix in this dataset, but after 100 rounds of running, the RMSE is still very large. My training set is 1000, the evaluation set is 200, and the test Set 200, could the RMSE reach a few thousand because the training set is too small?

**Comment by JUGGHM** [comment]
> alright, thank you very much! I trained the network today. I used the NYU-Depth v2 dataset and changed k to the camera's internal parameter matrix in this dataset, but after 100 rounds of running, the RMSE is still very large. My training set is 1000, the evaluation set is 200, and the test Set 200, could the RMSE reach a few thousand because the training set is too small?

You could refer to other works (e.g. NLSPN) for the training settings on NYU Depth V2 dataset. After a full training procedure, ENet could reach an RMSE of around 104 mm.

**Comment by yuyu19970716** [comment]
Good! Thank you so much! I'm going to try what you said right away!

**Comment by yuyu19970716** [comment]
Hi, I'm sorry to bother you again.
I have now trained PENet with KITTI, but I still have a question: can this network be trained with NYU depth v2? If not, I'll change my mind. Especially looking forward to your reply.
I changed the format of the NYU depth v2 dataset to the image format required by PENet.Can I do this? Because I ran into some other problems.
Very much looking forward to your reply!

**Comment by JUGGHM** [comment]
> Hi, I'm sorry to bother you again. I have now trained PENet with KITTI, but I still have a question: can this network be trained with NYU depth v2? If not, I'll change my mind. Especially looking forward to your reply. I changed the format of the NYU depth v2 dataset to the image format required by PENet.Can I do this? Because I ran into some other problems. Very much looking forward to your reply!

I didn't change the format of NYU depth V2 and directly rendered the NYU depth v2 dataloader from Park's (NLSPN) repositories. It works normally with a common training setting on the indoor dataset. I hope these files could help.
[nyu.json.txt](https://github.com/JUGGHM/PENet_ICRA2021/files/8631943/nyu.json.txt)
[nyu_loader.py.txt](https://github.com/JUGGHM/PENet_ICRA2021/files/8631944/nyu_loader.py.txt)

**Comment by yuyu19970716** [comment]
OK! I will take a serious look at what you have to offer!
thank you very much!
